src/vs/workbench/contrib/search/common/searchModel.ts

```diff
@@ -667,21 +819,113 @@ export class FolderMatch extends Disposable {
 	}
 }
 
+export class FolderMatchWithResource extends FolderMatch {
+
+	constructor(_resource: URI, _id: string, _index: number | null, _query: ITextQuery, _parent: SearchResult | FolderMatch, _searchModel: SearchModel, _closestRoot: FolderMatchWorkspaceRoot | null,
+		@IReplaceService replaceService: IReplaceService,
+		@IInstantiationService instantiationService: IInstantiationService,
+		@ILabelService labelService: ILabelService,
+		@IUriIdentityService uriIdentityService: IUriIdentityService
+	) {
+		super(_resource, _id, _index, _query, _parent, _searchModel, _closestRoot, replaceService, instantiationService, labelService, uriIdentityService);
+	}
+
+	override get resource(): URI {
+		return this._resource!;
+	}
+}
+
+/**
+ * FolderMatchWorkspaceRoot => folder for workspace root
+ */
+export class FolderMatchWorkspaceRoot extends FolderMatchWithResource {
+	constructor(_resource: URI, _id: string, _index: number, _query: ITextQuery, _parent: SearchResult, _searchModel: SearchModel,
+		@IReplaceService replaceService: IReplaceService,
+		@IInstantiationService instantiationService: IInstantiationService,
+		@ILabelService labelService: ILabelService,
+		@IUriIdentityService uriIdentityService: IUriIdentityService
+	) {
+		super(_resource, _id, _index, _query, _parent, _searchModel, null, replaceService, instantiationService, labelService, uriIdentityService);
+	}
+
+	private uriParent(uri: URI): URI {
+
+		let newUri = this.uriIdentityService.extUri.dirname(uri);
+
+		if (this.uriIdentityService.extUri.isEqual(newUri, uri)) {
+			// if parent is root, then the uri identity service dirname will return the same URI as entered.
+			newUri = URI.from({ scheme: uri.scheme, authority: uri.authority, path: '/', query: uri.query, fragment: uri.fragment });
```

Comment:
Make sure you don't forget to clean this up. But also you can do `uri.with({ path: '/' })`